### PR TITLE
Enhancing nebula to generate osquery conf from yaml files

### PR DESCRIPTION
**Pull request by praksinha**
Following changes are made:

- Added support to dynamically generate osquery conf in JSON format using topfile. One can now create host specific osquery conf using grains
- Added support to dynamically generate osquery flagfile using topfile. One can now create host specific flagfile using grains. This can be especially helpful if we want to enable/disable certain setting on certains host(s) or enable/disable extensions on few hosts.
- Added buildinfo in grains

### CODE REVIEW

File: hubblestack/extmods/modules/nebula_osquery.py [hunk 1]
```diff
@@ -615,6 +607,129 @@ def _get_top_data(topfile):
...
+                for key in flags_data:
+                    propdata = "--" + key + "=" + flags_data.get(key) + "\n"
+                    cf.write(propdata)
+        except Exception as e:
+            log.error("Failed to generate osquery flags file using topfile {0}".format(e))
+    
+    return flagfile
+
+
+def _get_osquery_top_data(topfile):
```
Comment (basepi) [inline hubblestack/extmods/modules/nebula_osquery.py]: Can we re-use (might need slight modifications) the existing `_get_top_data()` function? Seems like the topfiles are similar enough we can make it work without so much duplication.

Comment (praksinha) [reply hubblestack/extmods/modules/nebula_osquery.py]: Done.

**Comment by praksinha** [comment]
> This looks good to me. Are we waiting on any other changes/approvals?

I'm just testing this code on few supported platforms. This should be completed by EOD today. Once done, we can merge this.

**Comment by praksinha** [comment]
> This looks good to me. Are we waiting on any other changes/approvals?

Testing is complete on both linux and Windows. If there are no further review comments, we can merge this PR.

**Pull request merged**